pkg/schedule/schedule.go

```diff
@@ -97,6 +100,12 @@ func (s *Schedule) UnmarshalJSON(data []byte) error {
 	if err != nil {
 		return err
 	}
+
+	now := time.Now().UTC()
+	if startDateTime.Before(now) {
+		return ErrStartDateTimeInPast
+	}
```

Comment:
This should be validated in `addRuleReq`: https://github.com/absmach/magistrala/blob/35c84741d51600977bdd0a5b8b2c1798a5809089/re/api/requests.go#L23